zerver/webhooks/slack/tests.py

```diff
@@ -1,99 +1,265 @@
+
+    @override
+    def tearDown(self) -> None:
+        self.get_slack_api_data_patcher.stop()
+        self.check_slack_token_patcher.stop()
+        super().tearDown()
+
+    def mocked_get_slack_api_data(
+        self, url: str, get_param: str, token: str, **kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        slack_api_endpoints: Dict[str, Any] = {
+            "https://slack.com/api/users.info": {"name": USER},
+            "https://slack.com/api/conversations.info": {"name": CHANNEL},
+        }
+        self.assertIn(url, slack_api_endpoints)
+        return slack_api_endpoints[url]
+
     def test_slack_only_stream_parameter(self) -> None:
+        expected_message = EXPECTED_MESSAGE.format(user=USER, message=MESSAGE_WITH_NORMAL_TEXT)
         self.check_webhook(
-            "message_info",
+            "message_with_normal_text",
             EXPECTED_TOPIC,
-            EXPECTED_MESSAGE,
-            content_type="application/x-www-form-urlencoded",
+            expected_message,
+            content_type="application/json",
         )
 
     def test_slack_with_user_specified_topic(self) -> None:
         self.url = self.build_webhook_url(topic="test")
         expected_topic_name = "test"
+        expected_message = EXPECTED_MESSAGE.format(user=USER, message=MESSAGE_WITH_NORMAL_TEXT)
         self.check_webhook(
-            "message_info",
+            "message_with_normal_text",
             expected_topic_name,
-            EXPECTED_MESSAGE,
-            content_type="application/x-www-form-urlencoded",
+            expected_message,
+            content_type="application/json",
         )
 
     def test_slack_channels_map_to_topics_true(self) -> None:
         self.url = self.build_webhook_url(channels_map_to_topics="1")
-        expected_topic_name = "channel: general"
+        expected_message = EXPECTED_MESSAGE.format(user=USER, message=MESSAGE_WITH_NORMAL_TEXT)
+        expected_topic_name = TOPIC_WITH_CHANNEL.format(channel=CHANNEL)
         self.check_webhook(
-            "message_info",
+            "message_with_normal_text",
             expected_topic_name,
-            EXPECTED_MESSAGE,
-            content_type="application/x-www-form-urlencoded",
+            expected_message,
+            content_type="application/json",
         )
 
     def test_slack_channels_map_to_topics_true_and_user_specified_topic(self) -> None:
         self.url = self.build_webhook_url(topic="test", channels_map_to_topics="1")
+        expected_message = EXPECTED_MESSAGE.format(user=USER, message=MESSAGE_WITH_NORMAL_TEXT)
         expected_topic_name = "test"
         self.check_webhook(
-            "message_info",
+            "message_with_normal_text",
             expected_topic_name,
-            EXPECTED_MESSAGE,
-            content_type="application/x-www-form-urlencoded",
+            expected_message,
+            content_type="application/json",
```

Comment:
I think it would be better to use the `expected_topic_name` variable in the `build_webhook_url` function call as well; to indicate that the user-entered topic and the topic in which the message is to sent are the same.